packages/mui-material-next/src/ButtonGroup/ButtonGroup.types.ts

```diff
@@ -24,7 +24,7 @@ export interface ButtonGroupOwnProps {
    * @default 'primary'
    */
   color?: OverridableStringUnion<
-    'inherit' | 'primary' | 'secondary' | 'error' | 'info' | 'success' | 'warning',
+    'primary' | 'secondary' | 'tertiary',
```

Comment:
We shouldn't remove the `inherit` and other color options

Reply:
@DiegoAndai I had removed these because the other colors were also removed in the button. https://github.com/mui/material-ui/blob/master/packages/mui-material-next/src/Button/Button.types.ts#L40

Reply:
True, may I ask you to bring those back? We can do it in this PR

Reply:
@DiegoAndai `'error' | 'info' | 'success' | 'warning'` are no problem to implement, but `'inherit'` is very expensive to implement, because at every point where the property `color` is called, a check must first be made to ensure that it is not `inherit` and then the special forms for `inherit` must be implemented. So the question is whether you really want to keep `inherit` and whether it is worth the effort?

Reply:
That's a good point; it does seem like a specific use case that bloats the styles—tagging @mui/core to get more opinions.

Reply:
I discussed with some Core members and realized that the `inherit` color is pretty useful when dealing with composed components like navigation bars, modals, and alerts.

Is there a way we can improve the implementation so it's not so "expensive"?

Reply:
Okay, that sounds logical.
However, due to the complexity, I would suggest implementing the whole thing in a new PR, in which the missing colors are then returned for both the button and the button group. Here you could possibly also solve the problems with the color change of the button, which occurs when you use `filledTonal` or `elevated` as a variant and is only used `primary` as a color.